handlers/create_lease.go

```diff
@@ -68,33 +66,39 @@ func CreateLease(
 			return
 		}
 
-		unusedCluster, unusedClusterErr := findUnusedGKECluster(clusterMap, leaseMap)
-		uuidAndLease, expiredLeaseErr := findExpiredLease(leaseMap)
-		if unusedClusterErr != nil && expiredLeaseErr != nil {
-			htp.Error(w, http.StatusConflict, "no available or expired clusters found")
-			return
-		}
-		var availableCluster *container.Cluster
-		if unusedCluster != nil {
-			availableCluster = unusedCluster
-		}
-		if expiredLeaseErr == nil {
-			cl, ok := clusterMap.ClusterByName(uuidAndLease.Lease.ClusterName)
-			if !ok {
-				htp.Error(w, http.StatusInternalServerError, "cluster %s has an expired lease but does not exist in GKE", uuidAndLease.Lease.ClusterName)
+		availableCluster, err := searchForFreeCluster(clusterMap, leaseMap)
+		if err != nil {
+			switch e := err.(type) {
+			case errNoAvailableOrExpiredClustersFound:
+				htp.Error(w, http.StatusConflict, "no available clusters found")
```

Comment:
@vdice since this represents the barrier between the server & client, I wanted to hard-code something here. we could put the error text in parentheses, but in both cases here it says pretty much the same thing. anyway, would you prefer it?